docs/config.rst

```diff
@@ -0,0 +1,83 @@
+=====================
+Config Trait
+=====================
+
+.. php:trait:: ConfigTrait
+
+Agile Core implements support for read configuration files of different formats
+
+Introduction
+============
+
+This trait can be added to any object to load a configuration.
+Configuration files can be of 4 types : php, php-inline, json, yaml.
+
+Loading can be done in this way :
+
+    $object = new Object();
+    $object->readConfig('config.php','php');
+
+After loading, configuration elements can be retrieved in this way :
+
+    $object->getConfig('element_key','if not defined use this as default');
+
+if you need an element that is declared inside an array you can use a special syntax :
+
+    $object->getConfig('level1_array/level2_array/element_key','if not defined use this as default');
```

Comment:
i think we should also provide a good guideline on how 3rd party add-ons should use their class as a top level element key.

Reply:
Translator implementation for sure will need it and can be created as an addon.

At this moment the TranslatableTrait return translations already written in codebase, using an addon we can extract all this "tokens" and define it as an addon.

I think we probably need a composer script or a console command that create the folder in the project for you when add the addon.

next thursday we can speak about how can be implemented addons when need extra config, like we have done with actions, the results was really good.